components/collapsible/accordion/src/index.js

```diff
@@ -14,6 +14,11 @@ class CollapsibleAccordion extends Component {
     this._collapseItems(false, this.props.items.findIndex(item => !item.collapsed))
   }
 
+  componentWillReceiveProps (nextProps) {
+    const id = nextProps.items.findIndex(item => !item.collapsed)
+    this.setState({ openIndex: id === this.state.openIndex ? null : id })
```

Comment:
this code is repeated on line 26
You should refactor to segregate operations and reuse them.

Reply:
yes, i'm agree! done.